rasa/graph_components/providers/nlu_training_data_provider.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, Optional
+import os
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.importers.importer import TrainingDataImporter
+from rasa.shared.nlu.training_data.training_data import (
+    TrainingData,
+    DEFAULT_TRAINING_DATA_OUTPUT_PATH,
+)
+from rasa.shared.nlu.training_data.loading import load_data
+
+
+class NLUTrainingDataProvider(GraphComponent):
+    """Provides NLU training data during training and inference time."""
+
+    def __init__(
+        self,
+        model_storage: ModelStorage,
+        resource: Resource,
+        training_data: Optional[TrainingData] = None,
+    ) -> None:
+        """Creates NLU training data provider."""
+        self._model_storage = model_storage
+        self._resource = resource
+        self._training_data = training_data
+
+    @classmethod
+    def get_default_config(cls) -> Dict[Text, Any]:
+        """Get Default class configuration"""
+        return {"language": "en", "persist": False}
```

Comment:
I put `language` and `persist` in the configuration instead of passing them as arguments to the methods. Is this what we want?

Reply:
yep, that matches what Tobias requested. The persist option was a bit confusing for me at first - but I think it is here because importing the nlu data from disk can be a bit more costly than it looks (because we might have to parse the stories and extract the nlu data from there (cf. [importers here](https://github.com/RasaHQ/rasa/blob/2e82c9caab989348d57e74d918cc4286f85778a5/rasa/shared/importers/importer.py#L482)))

Reply:
and yep, the component should store those parameters in it's internal "config" and then use that config wherever needed (which makes sense for all parameters that e.g. don't depend on other components or on the training of other components - in contrast, the training data that is provided by this component will be expected as parameter in the method of another component but will not be passed as a config parameter)

Reply:
The reason persist is in the config is because it is controlled by a cli flag `--persist-nlu-data`. So it will change the config of the component, rather than come from a different component.

Is it's true it essentially means we bundle the nlu training data in with the model data in the model archive.

Reply:
I wonder if we should have a default language though 🤔

Reply:
So we just need to persist during "provide()" and not think about anything else, right? (Cause users could just switch of that flag to persist that data again and again)

Reply:
> I wonder if we should have a default language though 🤔

I was reluctant about putting english as default. I can remove it.